live.tsx

```diff
@@ -144,27 +156,95 @@ export function createLiveHandler<LoaderData = LivePageData>(
       }
 
       start("fetch-page-data");
-      let loaderData = undefined;
-
-      try {
-        if (typeof loader === "function") {
-          loaderData = await loader(req, ctx);
-        } else {
-          loaderData = await loadLiveComponents(
-            req,
-            ctx,
-            options as LoadLiveComponentsOptions,
-          );
+      const pageData = await loadLiveComponents(
+        req,
+        ctx,
+        options as LoadLiveComponentsOptions,
+      );
+      end("fetch-page-data");
+
+      const isLoaderProp = (value: any): value is string =>
+        typeof value === "string" && value.charAt(0) === "{" &&
+        value.charAt(value.length - 1) === "}";
+
+      // map back components from database to components for the editor, merging loader props into component props
+      const editorComponents = pageData.components.map((componentData) => {
+        if (
+          Object.values(componentData.props ?? {}).some(isLoaderProp)
+        ) {
+          const newComponentData = structuredClone(componentData);
+
+          for (
+            const [propName, value] of Object.entries(newComponentData.props)
+          ) {
+            if (isLoaderProp(value)) {
+              const loaderName = value.substring(1, value.length - 1);
```

Comment:
Extract this logic into a function/construct that possesses all the logic regarding this. E.g:

```tsx
const loaderWrapper = {
 isLoaderProp: (value: string) => value.startsWith('{') && value.endsWith('}'),
 toPropName: (loaderName: string) => `{${loaderName}}`,
 toLoaderName: (propValue: string) => propValue.substring(1, propValue.length - 1);
}